astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -24,211 +30,588 @@ class NDArithmeticMixin(object):
     so that the subclass sees NDData as the main superclass. See
     `~astropy.nddata.NDDataArray` for an example.
     """
-    def _arithmetic(self, operand, propagate_uncertainties, name, operation):
+
+    def arithmetic(self, operation, operand,
+                   uncertainty_correlation=0, meta_kwds_operate=None,
+                   meta_kwds_set=None, handle_uncertainties=True,
+                   handle_mask=True, handle_meta=True, compare_wcs=True):
         """
-        {name} another dataset (``operand``) to this dataset.
+        Base method which calculates the result of the arithmetic operation.
+
+        This class determines the result of the arithmetic operation on the
+        ``data`` with respect to their units and then forwards to other methods
+        to calculate the other properties for the result (like uncertainty).
 
         Parameters
         ----------
-        operand : `~astropy.nddata.NDData`
-            The second operand in the operation a {operator} b
-
-        propagate_uncertainties : bool
-            Whether to propagate uncertainties following the propagation rules
-            defined by the class used for the ``uncertainty`` attribute.
+        operation: `str`
+            The operation that is performed on the `NDData`. Supported are
+            ``addition``, ``subtraction``, ``multiplication`` and ``division``.
+
+        operand: `NDData` instance or something that can be converted to one.
+            The second operand in the operation.
+
+        uncertainty_correlation: ``Number`` or `~numpy.ndarray`
+            The correlation (rho) is defined between the uncertainties in
+            ``sigma_AB = sigma_A * sigma_B * rho`` (Latex?). If
+            ``handle_uncertainties`` is *not* ``True`` this will be ignored. A
+            value of ``0`` means uncorrelated (which is also the default)
+
+        meta_kwds_operate: ``None`` or `list`
```

Comment:
I think we're probably going to want to rename or simplify some of these arguments. I haven't had a close look yet though.